web/api/v1/output.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2020 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/promql"
+)
+
+const (
+	limitParam = "limit"
+	sortParam  = "sort"
+)
+
+func validateModifiers(r *http.Request) error {
+	params := r.URL.Query()
+	if p, ok := params[limitParam]; ok {
+		if l := len(p); l != 1 {
+			return fmt.Errorf("limit only takes 1 argument, got %d", l)
+		}
+		_, err := strconv.ParseUint(p[0], 10, 64)
+		if err != nil {
+			return err
+		}
+	}
+	if p, ok := params[sortParam]; ok {
+		for _, s := range p {
+			// Remove heading - which means deschending order.
+			s := strings.TrimLeft(s, "-")
+			if s == "1" {
+				// Sort by value.
+				continue
+			}
+			if !model.LabelNameRE.MatchString(s) {
+				return errors.Errorf("invalid label name: %q", s)
+			}
+		}
+	}
+	return nil
+}
+
+func applyModifiers(r *http.Request, res *promql.Result) (*promql.Result, error) {
+	var err error
+	params := r.URL.Query()
+	if p, ok := params[limitParam]; ok {
```

Comment:
What about sort? If we stick to `parse() return modifiers` it will be more verbose if we will forget to implement one modifier (e.g sort)